### DISCUSSION TITLE
Restrict `libp2p_ping::handler::Failure::Other` with `Sync`?

### DISCUSSION THREAD

**Discussion by drHuangMHT** [libp2p/rust-libp2p#5040] [Ideas]
`Other` variant holds value of this type:
https://github.com/libp2p/rust-libp2p/blob/4260702f749d45fb53ea7edd2fb0a728afb83488/protocols/ping/src/handler.rs#L97-L99
Which is not `Sync`, thus cannot be shared between threads, preventing it from being broadcast to subscribers.
Should we add `Sync` bound to it?

**Comment by thomaseizinger** [comment]
Yes that makes sense. I am wondering if this is a breaking change? I don't think it is so because we are _adding_ a trait bound, not removing one.
